File: gestaolegal/models/caso.py
from dataclasses import dataclass
from datetime import datetime
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from gestaolegal.models.arquivo_caso import ArquivoCaso
    from gestaolegal.models.atendido import Atendido
    from gestaolegal.models.processo import Processo
    from gestaolegal.models.user import UserInfo


@dataclass
class Caso:
    id_usuario_responsavel: int
    area_direito: str
    data_criacao: datetime
    id_criado_por: int
    situacao_deferimento: str
    status: bool

    id: int | None = None
    sub_area: str | None = None
    id_orientador: int | None = None
    id_estagiario: int | None = None
    id_colaborador: int | None = None
    data_modificacao: datetime | None = None
    id_modificado_por: int | None = None
    justif_indeferimento: str | None = None
    descricao: str | None = None
    numero_ultimo_processo: int | None = None
    usuario_responsavel: "UserInfo | None" = None
    clientes: list["Atendido"] | None = None
    orientador: "UserInfo | None" = None
    estagiario: "UserInfo | None" = None
    colaborador: "UserInfo | None" = None
    criado_por: "UserInfo | None" = None
    modificado_por: "UserInfo | None" = None
    processos: list["Processo"] | None = None
    arquivos: list["ArquivoCaso"] | None = None

File: web/src/lib/types/caso.ts
import type { User } from './user';
import type { Atendido } from './atendido';
import type { Processo } from './processo';

export interface Caso {
	id: number;
	id_usuario_responsavel: number;
	usuario_responsavel: User;
	area_direito: string;
	sub_area: string | null;
	clientes: Atendido[];
	id_orientador: number | null;
	orientador: User | null;
	id_estagiario: number | null;
	estagiario: User | null;
	id_colaborador: number | null;
	colaborador: User | null;
	data_criacao: string;
	id_criado_por: number;
	criado_por: User;
	data_modificacao: string | null;
	id_modificado_por: number | null;
	modificado_por: User | null;
	situacao_deferimento: string;
	justif_indeferimento: string | null;
	status: boolean;
	descricao: string | null;
	numero_ultimo_processo: number | null;
	processos?: Processo[];
	arquivos?: ArquivoCaso[];
}

export interface ListCaso {
	id: number;
	area_direito: string;
	sub_area: string | null;
	situacao_deferimento: string;
	status: boolean;
	descricao: string | null;
	data_criacao: string;
	usuario_responsavel: User;
	clientes: Atendido[];
}

export interface ArquivoCaso {
	id: number;
	link_arquivo: string;
}
